docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+        summary = "lorem ipsum";
+        action = opt variant {
+            ManageNervousSystemParameters: record {
+                default_followees : opt record { 
+                    followees : vec record { 
+                                nat64; 
+                                record { followees : vec record { id : vec {1:nat8, 2:nat8} } }  
+                    } 
+                };
+                max_dissolve_delay_seconds = opt 42:nat64;
+                max_dissolve_delay_bonus_percentage = opt 42:nat64;
+                max_followees_per_function = opt 42:nat64;
+                neuron_claimer_permissions = opt record { permissions = vec {1:int32, 2:int32} };
+                neuron_minimum_stake_e8s = opt 42:nat64;
+                max_neuron_age_for_age_bonus = opt 42:nat64;
+                initial_voting_period_seconds = opt 42:nat64;
+                neuron_minimum_dissolve_delay_to_vote_seconds = opt 42:nat64;
+                reject_cost_e8s = opt 42:nat64;
+                max_proposals_to_keep_per_action = opt 42:nat32;
+                wait_for_quiet_deadline_increase_seconds = opt 42:nat64;
+                max_number_of_neurons = opt 42:nat64;
+                transaction_fee_e8s = opt 42:nat64;
+                max_number_of_proposals_with_ballots = opt 42:nat64;
+                max_age_bonus_percentage = opt 42:nat64;
+                neuron_grantable_permissions = opt record { permissions : vec {1:int32, 2:int32} };
+                voting_rewards_parameters = opt record {
+                    final_reward_rate_basis_points = opt 42:nat64;
+                    initial_reward_rate_basis_points = opt 42:nat64;
+                    reward_rate_transition_duration_seconds = opt 42:nat64;
+                    round_duration_seconds = opt 42:nat64;
+                };
+                maturity_modulation_disabled = opt true:bool;
+                max_number_of_principals_per_neuron = opt 42:nat64;
+            };
+    }
+)'
+```
+
+## `AddGenericNervousSystemFunction`
+
+### Relevant type signatures
+
+```candid
+    type AddGenericNervousSystemFunction : NervousSystemFunction;
+
+    type NervousSystemFunction = record {
+        id : nat64;
+        name : text;
+        description : opt text;
+        function_type : opt FunctionType;
+    };
+
+    type FunctionType = variant {
+        NativeNervousSystemFunction : record {};
+        GenericNervousSystemFunction : GenericNervousSystemFunction;
+    };
+
+    type GenericNervousSystemFunction = record {
+        validator_canister_id : opt principal;
+        target_canister_id : opt principal;
+        validator_method_name : opt text;
+        target_method_name : opt text;
+    };
+```
+
+### Putting it together
+
+```candid
+    type AddGenericNervousSystemFunction: record {
+        id : nat64;
+        name : text;
+        description : opt text;
+        function_type : opt variant {
+            NativeNervousSystemFunction : record {};
```

Comment:
since this is a variant, I would expect that we only have here GenericNervousSystemFunction or Native.. maybe good to double check with an engineer though
```suggestion

```